components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.webclient.actions.js

```diff
@@ -382,9 +381,13 @@ OME.refreshThumbnails = function(options) {
     var $thumbs = $(spw_selector + ", " + search_selector);
     if ($thumbs.length > 0){
         $thumbs.each(function(){
-            var $this = $(this),
-                base_src = $this.attr('src').split('?')[0];
-            $this.attr('src', base_src + "?_="+rdm);
+            var _t = $(this);
+                OME.load_thumbnail(
+                    options.imageId, options.thumbnail_url,
+                    function(thumb) {
+                        _t.attr('src', thumb);
+                    }
+                );
```

Comment:
Instead of being passed the full ```thumbnail_url```, ```refreshThumbnails()``` could be passed the ```/webgateway/``` base url and then it could construct the ```/webgateway/get_thumbnails/``` OR ```/webgateway/get_thumbnail/id/``` urls itself, depending on whether it was refreshing ALL thumbs or just a single image.
Alternatively, we could add the ```/webgateway/``` url to the global WEBCLIENT.URLS dictionary, if we know that this is always going to be available, similar to WEBCLIENT.URLS.webindex which we use in ```ome.tree.js```